opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/b3_format.py

```diff
@@ -95,19 +97,34 @@ def extract(
                 or flags
             )
 
+        if (
+            fullmatch(r"[\da-fA-F]{16}|[\da-fA-F]{32}", trace_id) is None
+            or fullmatch(r"[\da-fA-F]{16}", span_id) is None
+        ):
+            trace_id = generate_trace_id()
+            span_id = generate_span_id()
+            sampled = "0"
+
         options = 0
         # The b3 spec provides no defined behavior for both sample and
         # flag values set. Since the setting of at least one implies
         # the desire for some form of sampling, propagate if either
         # header is set to allow.
         if sampled in self._SAMPLE_PROPAGATE_VALUES or flags == "1":
             options |= trace.TraceFlags.SAMPLED
+
+        if isinstance(trace_id, str):
+            trace_id = int(trace_id, 16)
+
+        if isinstance(span_id, str):
+            span_id = int(span_id, 16)
```

Comment:
Looks like you know they are strings in the regex checks on L100, so if that is the case this can just go in an `else` block on L107 without the `isinstance` checks?

Reply:
Fixed